io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,219 @@
+#include <pcl/io/eigen.h>
+#include <pcl/io/eigen.h>
+#include <pcl/io/boost.h>
+#include <pcl/io/grabber.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+#include <librealsense2/rs.hpp>
+#include <pcl/io/real_sense_2_grabber.h>
+
+namespace pcl
+{
+
+RealSense2Grabber::RealSense2Grabber()
+: running(false)
+, quit(false)
+, signal_PointXYZ(nullptr)
+, signal_PointXYZI(nullptr)
+, signal_PointXYZRGB(nullptr)
+, signal_PointXYZRGBA(nullptr)
+{
+    
+    // for now we only create the xyzrgb signal
+    signal_PointXYZ = createSignal<signal_librealsense_PointXYZ>();
+    signal_PointXYZI = createSignal<signal_librealsense_PointXYZI>();
+    signal_PointXYZRGB = createSignal<signal_librealsense_PointXYZRGB>();
+    signal_PointXYZRGBA = createSignal<signal_librealsense_PointXYZRGBA>();
+}
+
+
+RealSense2Grabber::~RealSense2Grabber()
+{
+    stop();
+
+    disconnect_all_slots<signal_librealsense_PointXYZ>();
+    disconnect_all_slots<signal_librealsense_PointXYZI>();
+    disconnect_all_slots<signal_librealsense_PointXYZRGB>();
+    disconnect_all_slots<signal_librealsense_PointXYZRGBA>();
+
+    thread.join();
+
+    pipe.stop();    
+}
+
+void RealSense2Grabber::start()
+{
+    running = true;
+
+	rs2::config cfg;
+
+	cfg.enable_stream(RS2_STREAM_COLOR, 424, 240, RS2_FORMAT_RGB8, 30);
+	cfg.enable_stream(RS2_STREAM_DEPTH, 424, 240, RS2_FORMAT_ANY, 30);
+	cfg.enable_stream(RS2_STREAM_INFRARED, 424, 240, RS2_FORMAT_ANY, 30);
```

Comment:
I think that it should provide some way to set resolution and frame rate.

Reply:
This can now be updated:

https://github.com/complexrehabsystems/pcl/blob/e342155a33a8a9aa2cf3322e8c74782e8d88b6c3/io/src/real_sense_2_grabber.cpp#L51